src/App.tsx

```diff
@@ -8,37 +8,114 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getUsers } from './services/user';
+import { getUserPosts } from './services/post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(false);
+
+  const loadUsers = async () => {
+    try {
+      const loadedUsers = await getUsers();
+
+      setUsers(loadedUsers);
+    } catch {
+      throw new Error('Failed to load users');
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)

you may use `setErrorMessage`, but show another message, etc.